### ISSUE TITLE
[Bug] TabBarBackgroundColor of ShellContent not instantly updating upon UserAppTheme changed

### ISSUE THREAD

**Issue by Cfun1** [xamarin/Xamarin.Forms#13123]
### Description
When changing `UserAppTheme`, the styled `TabBarBackgroundColor ` property with `AppThemeBinding` is not changing instantly but only when a navigation action is done.

### Steps to Reproduce
1.     <Application.Resources>
        <Style TargetType="{x:Type ShellContent}">
            <Setter Property="Shell.TabBarBackgroundColor" Value="{AppThemeBinding Dark=red, Light=yellow}"/>
        </Style>
        </Application.Resources>

2. Change `UserAppTheme` from light to dark or opposite.

### Expected Behavior
Color of `TabBarBackgroundColor`  takes effect instantly when `UserAppTheme` changes.
### Actual Behavior
Color of `TabBarBackgroundColor` takes effect only after navigating in top tabs or bottom tabs (hanging the page).
### Basic Information

- Version with issue: 5.0.0.1791-pre5.
- Platform Target Frameworks: Tested on android only.
- Android Support Library / AndroidX Version: Android 10.0 API 29

**Comment by rachelkang** [comment]
@Cfun1 thanks for submitting this issue! It seems to be a duplicate of https://github.com/xamarin/Xamarin.Forms/issues/12807 - can you confirm this?

**Comment by Cfun1** [comment]
@rachelkang my issue is not about animation neither during app startup but during running, although I cannot confirm if the root cause of theses two issues is common, it could be.

**Comment by jsuarezruiz** [comment]
Can reproduce the issue:
[Issue13123.zip](https://github.com/xamarin/Xamarin.Forms/files/5943841/Issue13123.zip)
![issue13123](https://user-images.githubusercontent.com/6755973/107223915-0e33cd00-6a17-11eb-82a2-f4657773ecc3.gif)

**Comment by ChristianFrom** [comment]
Same with iOS.
